data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt

```diff
@@ -0,0 +1,56 @@
+package com.ait.dataimplementation
+
+import com.ait.dataapi.AppIdProvider
+import com.ait.dataapi.ForecastService
+import com.ait.dataapi.model.ForecastSource
+import com.ait.dataapi.model.ForecastedWeatherInfo
+import com.ait.dataimplementation.service.openWeather.OpenWeather
+import com.ait.dataimplementation.service.openWeather.OpenWeatherService
+import com.ait.dataimplementation.service.template.TemplateDataService
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+
+class WeatherForecastManager(private val appIdProvider: AppIdProvider, private val source: ForecastSource) {
+
+    suspend fun getWeatherInfo(latitude: Double, longitude: Double): ForecastedWeatherInfo? {
+
+        // TODO: Will be reworked later with Hilt
+        val forecastService: ForecastService = when (source) {
+            ForecastSource.OPEN_WEATHER -> {
+                val client = OkHttpClient().newBuilder()
```

Comment:
There must be another class for api building. Just pass ForecastService interface in constructor

Reply:
We must follow single responsibility rule